wagtail/admin/templates/wagtailadmin/tables/table.html

```diff
@@ -1,6 +1,9 @@
 {% load wagtailadmin_tags %}
 
 <table{% include "wagtailadmin/shared/attrs.html" with attrs=table.attrs %}>
+    {% if table.has_caption %}
```

Comment:
As per comment on methods, let's keep this simple and avoid too much indirection. 

Having the method call like this should work. 

```suggestion
    {% with caption=table.get_caption %}
        {% if caption %}
          <caption...
        {% endif %}
{% endwith %}
```

Reply:
@lb-  just for the understanding point of view, may I ask why are we going with context manager based rendering ? rather than rendering with a direct if block?

Reply:
I would not necessarily call this a context manager (but maybe I am wrong). This really just creates a variable alias and avoids us calling `get_caption` multiple times and also avoids the complexity overhead of additional methods on the Table class.

https://docs.djangoproject.com/en/5.0/ref/templates/builtins/#with

In theory we could just render an empty caption (as it's not visible) but this is probably not the right way to do thins semantically.